assets/helpers/checkoutForm/checkoutForm.js

```diff
@@ -50,6 +50,29 @@ export const formInputs = (formClassName: string): Array<HTMLInputElement> => {
 export const getForm: string => Object | null =
   (formName: string) => document.querySelector(`.${formName}`);
 
+const getInvalidReason = (validityState: ValidityState) => {
+  if (validityState.valueMissing) {
+    return '-value-missing';
+  } else if (validityState.patternMismatch) {
+    return '-pattern-mismatch';
+  }
+  return '';
+};
+
+export const invalidReason = (form: Object | null): string => {
+  let invalidReasonString = '';
```

Comment:
maybe we should wrap this whole thing in a try/catch and also log an event on error, just so we know when the whole constraint validation API is missing in a particular browser.

Reply:
done